src/app/panels/terminal.js

```diff
@@ -437,10 +432,10 @@ class Terminal extends Plugin {
     self._shell('remix.help()', self.commands, () => {})
     self.commands.html(intro)
 
-    self._components.txLogger = new TxLogger(self._opts.eventsDecoder, self._opts.txListener, this, self.blockchain)
-    self._components.txLogger.event.register('debuggingRequested', (hash) => {
+    self._components.txLogger = new TxLogger(this, self.blockchain)
+    self._components.txLogger.event.register('debuggingRequested', async (hash) => {
       // TODO should probably be in the run module
-      if (!self._opts.appManager.isActive('debugger')) self._opts.appManager.activateOne('debugger')
+      if (!await self._opts.appManager.isActive('debugger')) await self._opts.appManager.activatePlugin('debugger')
```

Comment:
You can do : 
```typescript
await this.call('manager', 'activatePlugin', 'debugger')
```
Like that you don't need to pass the dependency on appManager.

In any case, `activatePlugin` always verify if the plugin is activated first. So you don't need the if statement.